include/benchmark/benchmark_api.h

```diff
@@ -247,34 +247,39 @@ enum BigO {
   oNCubed,
   oLogN,
   oNLogN,
-  oAuto
+  oAuto,
+  oLambda
 };
 
+// BigOFunc is passed to a benchmark in order to specify the asymptotic 
+// computational complexity for the benchmark.
+typedef double(BigOFunc)(size_t);
```

Comment:
`std::function<double(size_t)>` doesn't compile on the target platforms?

also, there's no need for n to be size_t. it isn't the length of a container, so int is preferable.